spec/solver_spec.rb

```diff
@@ -0,0 +1,47 @@
+require_relative '../tdd/solver'
+
+describe Solver do
+  describe '#factorial' do
+    it 'returns 1 for 0' do
+      solver = Solver.new
+      expect(solver.factorial(0)).to eq(1)
+    end
+  end
```

Comment:
- [ ] As required for the project, kindly ensure that the `#factorial` method also has test for **Negative value which raises an exception as an expectation** and **Special case - when input is 0, the result should equal 1**